code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm

```diff
@@ -591,19 +591,21 @@
 	name = "Laughter"
 	id = "laughter"
 	description = "Some say that this is the best medicine, but recent studies have proven that to be untrue."
-	metabolization_rate = INFINITY
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
 	color = "#FF4DD2"
 
 /datum/reagent/consumable/laughter/on_mob_life(mob/living/M)
-	M.emote("laugh")
-	var/laughnum = rand(1,2)
-	if(M.gender == MALE)
-		if(laughnum == 1)
+	if(!M.silent && prob(10))
+		M.emote("laugh")
+		var/laughsound = pick('sound/voice/manlaugh1.ogg', 'sound/voice/manlaugh2.ogg')
+		if(M.gender == MALE)
+			playsound(get_turf(M), laughsound, 50, 1)
+		else if(M.gender == FEMALE)
+			playsound(get_turf(M), 'sound/voice/womanlaugh.ogg', 65, 1)
+		else//non-binary gender just sounds like a man
 			playsound(get_turf(M), 'sound/voice/manlaugh1.ogg', 50, 1)
-		if(laughnum == 2)
-			playsound(get_turf(M), 'sound/voice/manlaugh2.ogg', 50, 1)
-	else if(M.gender == FEMALE)
-		playsound(get_turf(M), 'sound/voice/womanlaugh.ogg', 65, 1)
-	else//non-binary gender just sounds like a man
-		playsound(get_turf(M), 'sound/voice/manlaugh1.ogg', 50, 1)
+	if(holder.has_reagent("nothing"))
+		metabolization_rate = 5 * REAGENTS_METABOLISM
```

Comment:
nig that's gonna be *5ed for every on_mob_life check,that's probs not what you want,is it